core/src/summary.cc

```diff
@@ -3,18 +3,21 @@
 namespace prometheus {
 
 Summary::Summary(const Quantiles& quantiles,
-                 const std::chrono::milliseconds max_age, const int age_buckets)
-    : quantiles_{quantiles},
-      count_{0},
-      sum_{0},
-      quantile_values_{quantiles_, max_age, age_buckets} {}
-
-void Summary::Observe(const double value) {
+                 const std::chrono::milliseconds& max_age,
+                 const int& age_buckets)
+    : quantiles_(quantiles),
+      count_(0),
+      sum_(0),
+      quantile_values_(quantiles_, max_age, age_buckets) {}
+
+void Summary::Observe(const double value, const bool& alert) {
   std::lock_guard<std::mutex> lock(mutex_);
 
   count_ += 1;
   sum_ += value;
   quantile_values_.insert(value);
+  last_update_ = std::time(nullptr);
+  if (alert) AlertIfNoFamily();
```

Comment:
Well, clearly you wouldn't understand since your family is awesome!

Joking aside, prometheus keeps a list of metrics (families) for publishing to sockets (which eventually get added to the database). Now that we have the ability to expire metrics (remove them from their family), we want to alert the user if and when they are trying to use a metric that will not get published.